resources/vdir.rb

```diff
@@ -63,11 +65,13 @@
 end
 
 action :add do
-  if !@current_resource.physical_path
+  if exists
+    Chef::Log.debug("#{new_resource} virtual directory already exists - nothing to do")
+  else
     converge_by "Created the VDIR - \"#{new_resource}\"" do
       cmd = "#{appcmd(node)} add vdir /app.name:\"#{vdir_identifier}\""
       cmd << " /path:\"#{new_resource.path}\""
-      cmd << " /physicalPath:\"#{windows_cleanpath(new_resource.physical_path)}\""
```

Comment:
no need to clean windows path since we clean windows file path in `load_current_value`